backend/app/adapter/db/userurl.go

```diff
@@ -32,6 +32,35 @@ VALUES ($1,$2)
 	return err
 }
 
+// FindAliasesByUser fetches all the aliases corresponding to the given user.
+// TODO(issue#the_issue_number): allows API client to filter urls based on visibility.
+func (u UserURLRelationSQL) FindAliasesByUser(user entity.User) ([]string, error) {
+	statement := fmt.Sprintf(`SELECT "%s" FROM "%s" WHERE "%s"=$1;`,
+		table.UserURLRelation.ColumnURLAlias,
+		table.UserURLRelation.TableName,
+		table.UserURLRelation.ColumnUserEmail,
+	)
+
+	var aliases []string
+	rows, err := u.db.Query(statement, user.Email)
+	if err != nil {
+		return aliases, nil
+	}
+	defer rows.Close()
```

Comment:
We need to move this line to between line 45 and line 46 to ensure we properly close database connection before returning at line 47.